app/src/main/java/com/simformsolutions/sample/app/ui/main/MainViewModel.kt

```diff
@@ -34,7 +38,21 @@ class MainViewModel @Inject constructor(
     private val mainRepository: MainRepository
 ) : ViewModel() {
 
-    val repositories = Pager(
+    private val _isRefreshing = MutableStateFlow(false)
+    val isRefreshing = _isRefreshing.asStateFlow()
+
+    var repositories = getAllRepositories()
+
+    fun refresh() {
+        viewModelScope.launch {
+            _isRefreshing.value = true
+            repositories = getAllRepositories()
+            delay(1000)
```

Comment:
Delay to display it's loading is not a good practice. If page loads fast it's great.

Reply:
Updated to reset the value of `_isRefreshing` on list loaded rather than on delay.